src/API/Helpers/State/RedisStateHandler.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace Auth0\SDK\API\Helpers\State;
+
+use Predis\Client;
+use Auth0\SDK\API\Helpers\State\StateHandler;
+
+/*
+ * This file is part of Auth0-PHP package.
+ *
+ * (c) Auth0
+ *
+ * For the full copyright and license information, please view the LICENSE file
+ * that was distributed with this source code.
+ */
+
+/**
+ * Redis based implementation of StateHandler.
+ *
+ * @author Auth0
+ */
+class RedisStateHandler implements StateHandler
+{
+    const STATE_NAME = 'webauth_state';
+
+    private $cache;
+
+    /**
+     * Auth0RedisStateHandler constructor.
+     *
+     * @param Client $cache
+     */
+    public function __construct(Client $cache)
+    {
+        $this->cache = $cache;
+    }
+
+    /**
+     * Generate state value to be used for the state param value during authorization.
+     *
+     * @return string
+     */
+    public function issue()
+    {
+        $state = uniqid('', true);
+        $this->store($state);
+        return $state;
+    }
+
+    /**
+     * Store a given state value to be used for the state param value during authorization.
+     *
+     * @param string $state
+     *
+     * @return mixed|void
+     */
+    public function store($state)
+    {
+        $this->cache->sadd(self::STATE_NAME, $state);
```

Comment:
I'm reading the Redis command this maps to:

https://redis.io/commands/sadd

... and it looks like calling this twice will store 2 values, rather than override. Is that correct? What happens if 10 people start the login process at around the same time and then return to the site, how is their specific state tied to their login session?